python/ray/dag/compiled_dag_node.py

```diff
@@ -916,7 +915,15 @@ def _preprocess(self) -> None:
         self.input_task_idx, self.output_task_idx = None, None
         self.actor_task_count.clear()
 
-        nccl_actors_p2p: Set["ray.actor.ActorHandle"] = set()
+        # Senders are the DAG nodes who use the custom NCCL group as their
+        # transport in the type hints. Receivers are downstream DAG nodes of
+        # the senders.
```

Comment:
Adding a comment on why both `custom_nccl_group_to_p2p_senders` and `custom_nccl_group_to_p2p_senders_and_receivers` need to exist might be helpful, but up to you

Reply:
Great point, will do!